### PR TITLE
[X86][StrictFP] Combine fcmp + select to fmin/fmax for some predicates

### PR DESCRIPTION
**Pull request by phoebewang**
X86 maxss/minss etc. instructions won't turn SNaN to QNaN, so we can combine fcmp + select to them for some predicates.

### CODE REVIEW

File: llvm/lib/Target/X86/X86ISelLowering.cpp [hunk 1]
```diff
@@ -46494,17 +46496,22 @@ static SDValue combineSelect(SDNode *N, SelectionDAG &DAG,
   // x<=y?x:y, because of how they handle negative zero (which can be
   // ignored in unsafe-math mode).
   // We also try to create v2f32 min/max nodes, which we later widen to v4f32.
-  if (Cond.getOpcode() == ISD::SETCC && VT.isFloatingPoint() &&
-      VT != MVT::f80 && VT != MVT::f128 && !isSoftF16(VT, Subtarget) &&
-      (TLI.isTypeLegal(VT) || VT == MVT::v2f32) &&
+  if ((Cond.getOpcode() == ISD::SETCC ||
+       Cond.getOpcode() == ISD::STRICT_FSETCC ||
```
[the remark this answers is not in the document]
Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:46500]: You are correct, removed.

**Comment by phoebewang** [comment]
ping?

**Review by RKSimon — APPROVED** [review]
LGTM - @andykaylor anything else?

**Review by andykaylor — APPROVED** [review]
lgtm

**Pull request merged**